File: src/com/gmail/marcosav2010/command/base/ConnectionKeyCMD.java
package com.gmail.marcosav2010.command.base;

import com.gmail.marcosav2010.command.Command;
import com.gmail.marcosav2010.main.Main;
import com.gmail.marcosav2010.peer.Peer;

class ConnectionKeyCMD extends Command {

	ConnectionKeyCMD() {
		super("connectionkey", new String[] { "ck", "ckey" }, "[peer]");
	}

	@Override
	public void execute(String[] arg, int args) {
		int pCount = Main.getInstance().getPeerManager().count();

		if (pCount > 1 && args == 0) {
			log.log("ERROR: Specify peer.");
			return;
		}

		boolean autoPeer = pCount == 1 && args == 0;

		Peer peer;

		if (!autoPeer) {
			String peerName = arg[0];

			if (Main.getInstance().getPeerManager().exists(peerName)) {
				peer = Main.getInstance().getPeerManager().get(peerName);
			} else {
				log.log("ERROR: Peer \"" + peerName + "\" doesn't exists.");
				return;
			}
		} else
			peer = Main.getInstance().getPeerManager().getFirstPeer();

		String k;

		k = peer.getConnectionManager().getHandshakeAuthentificator().getConnectionKeyString();

		log.log("-----------------------------------------------------");
		log.log("\tConnection Key => " + k);
		log.log("-----------------------------------------------------");
	}
}

File: cli/src/main/java/com/gmail/marcosav2010/command/base/ConnectionKeyCMD.kt
package com.gmail.marcosav2010.command.base

import com.gmail.marcosav2010.command.Command
import com.gmail.marcosav2010.main.Main
import com.gmail.marcosav2010.peer.Peer

internal class ConnectionKeyCMD : Command("connectionkey", arrayOf("ck", "ckey"), "[peer]") {

    override fun execute(arg: Array<String>, length: Int) {
        val peerManager = Main.instance.peerManager!!

        val pCount = peerManager.count()
        if (pCount > 1 && length == 0) {
            log.log("ERROR: Specify peer.")
            return
        }

        val autoPeer = pCount == 1 && length == 0
        val peer: Peer = if (!autoPeer) {
            val peerName = arg[0]
            if (peerManager.exists(peerName)) {
                peerManager[peerName]!!
            } else {
                log.log("ERROR: Peer \"$peerName\" doesn't exists.")
                return
            }
        } else peerManager.firstPeer

        val k: String
        k = peer.connectionManager.handshakeAuthenticator.connectionKeyString

        log.log("-----------------------------------------------------")
        log.log("\tConnection Key => $k")
        log.log("-----------------------------------------------------")
    }
}